### DISCUSSION TITLE
compile error

### DISCUSSION THREAD

**Discussion by itzc33** [cesanta/mongoose#2464] [Ask questions, get answers] [answered]
![2023-11-07_18-16-01](https://github.com/cesanta/mongoose/assets/150168632/75367a2d-ccb3-4f57-ba44-9e4ea3841c96)

![2023-11-07_18-17-29](https://github.com/cesanta/mongoose/assets/150168632/776932db-8ac8-43b4-8732-8cd76d3d4455)

please advise

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

If there is something that you can't do, please describe it with the smallest piece of code possible.

Please do not paste screenshots
Please describe clearly your problem

**_You are clearly missing an include file in whatever you are doing, that is a basic C preprocessor error._**
All our examples are continually built on a CI/CD system, so clearly you are not building one of our examples so something you wrote.

[the remark this answers is not in the document]
**Reply by scaprile** [reply]
### You are not including mongoose.h, please read the documentation, read the tutorials, do the tutorials, learn your tools, learn your programming language.

https://mongoose.ws/documentation/tutorials/stm32/cube/#6-integrate-mongoose
